### ISSUE TITLE
Make working directory for media transcoding configurable

### ISSUE THREAD

**Issue by bartbilliet** [BiglySoftware/BiglyBT#1686]
* OS and version: Windows 10 1809
* BiglyBT Version Number: 2.4.0.0

When clicking the Play Now button, the working directory for transcoding has been automatically set to an external drive. When the drive is unplugged, there is no way currently to use Play Now. 

![image](https://user-images.githubusercontent.com/23324404/84950572-6831fc00-b0ef-11ea-8d15-815a1f5a0ba8.png)

Please, add a configuration option to allow changing the transcoding directory for the Play Now feature. 

Code reference: https://github.com/BiglySoftware/BiglyBT/blob/master/core/src/com/biglybt/core/devices/impl/DeviceImpl.java#L953

**Comment by parg** [comment]
How about Tools->Options->Devices ?

You might have to delete a device and re-add it to make it pick up the change.

**Comment by bartbilliet** [comment]
Hmm, I didn't see that option in the configuration, thanks! 

The working directory was actually configured to a path on a local drive, which was accessible. However, the experience was that transcoding did not work until the external drive was attached. (see screenshot when drive not attached)

When reconfiguring to a different path, saving, and following reconfiguring the option back to the original path, the transcoding now seems to works. It looks like there could be a bug where internally Bigly keeps using the original path to the external drive, even though it's showing a new local path already in the configuration?

**Comment by parg** [comment]
Possibly, I believe that each device remembers its transcode location when created (from the default) - this can be seen by right-clicking on the device and selecting 'properties'

**Comment by bartbilliet** [comment]
I'm actually not sure what you mean - in the devices list on the left navigation I don't find the external drive (even it is connected). In the configuration options it looks like there is only a single configuration for the transcoding directory, not a device-specific one?

**Comment by parg** [comment]
Right-click on the device in the sidebar to show the context menu and select 'Properties' from it. It show's the device's specific transcode location,

**Comment by bartbilliet** [comment]
It appears the device (external hard drive - which is connected) is not listed in the devices sidebar. I might be misunderstanding the devices concept here though. 
Windows UPnP is listed in the devices sidebar, the transcode location seems to equal what is configured in the options.

**Comment by parg** [comment]
Ah, not the external hard drive, sorry for the confusion, I should have looked into this more before replying.

When analysing media for playback a device is used called 'Media Analyser' - if this is hidden then right-click on the 'Devices' entry in the sidebar and select 'show hidden devices' from the menu.

Then right-click on Media Analyser and select Properties to see where its transcode location is.

**Comment by bartbilliet** [comment]
Thanks, I did find the 'Media Analyser' now. The transcode Area is also located on the local drive, not the external drive, similar to what the configuration in 'Tools > Options > Devices' is showing. Although this might indeed originally still have pointed to the external drive (given that a few days ago I reconfigured to a different path, saving, and following reconfiguring the option back to the original path in the options)

I'm not sure why the transcoder kept using the external drive, while the Tools > Options > Devices was showing the local drive. The problem is fixed now, I was just trying to see if this could be a bug where the software doesn't properly show the transcoder path used in Tools > Options > Devices when the external device is unplugged.
